### DISCUSSION TITLE
should_launch_new_window + startup_commands

### DISCUSSION THREAD

**Discussion by NailGit** [ahrm/sioyek#823] [General]
In my prefs_user I use startup commands:  
`startup_commands toggle_custom_color;toggle_scrollbar;toggle_mouse_drag_mode;fit_to_page_width`  
I also set `should_launch_new_window 1` to open multiple documents in separate windows. But the new window is open with the default config and my startup commands are ignored. The first file is opened normally with my startup options. Is it possible to fix this problem?

**Comment by ahrm** [comment]
Which version of sioyek are you using? Can you share your `prefs_user.config` file?

**Comment by NailGit** [comment]
The version is 2.0.0, I compiled it myself by using GitHub.  I start binary sioyek in `sioyek/build/`. My prefs_user.config file is below 
```startup_commands toggle_custom_color;toggle_scrollbar;toggle_mouse_drag_mode;fit_to_page_width
custom_background_color 1. 0.996 0.812
custom_text_color       0 0 0
status_bar_color        0 0 0
status_bar_text_color   1 1 1
status_bar_font_size    25
font_size 30
page_separator_width 2
page_separator_color 0 0 0
highlight_color_g 0.824 0.969 0.835
highlight_color_y 0.941 0.933 0.541
highlight_color_r 0.927 0.718 0.639
highlight_color_b 0.678 0.808 0.941
should_launch_new_window 1
```

And my keys_user.config is 
```
fit_to_page_width ww
fit_to_page_width_smart WW
toggle_custom_color cc
toggle_scrollbar <C-b>
embed_annotations sa
```

**Reply by ahrm** [reply]
Did you delete the `startup_commands` line from the `prefs_user.config` that you posted above?

**Reply by NailGit** [reply]
Ah, sorry. This line was not copied. The whole content of this file is below 
```
startup_commands toggle_custom_color;toggle_scrollbar;toggle_mouse_drag_mode;fit_to_page_width
custom_background_color 1. 0.996 0.812
custom_text_color       0 0 0
status_bar_color        0 0 0
status_bar_text_color   1 1 1
status_bar_font_size    25
font_size 30
page_separator_width 2
page_separator_color 0 0 0
highlight_color_g 0.824 0.969 0.835
highlight_color_y 0.941 0.933 0.541
highlight_color_r 0.927 0.718 0.639
highlight_color_b 0.678 0.808 0.941
should_launch_new_window 1
```

**Reply by ahrm** [reply]
I am unable to reproduce this issue on my windows system. what operating system are you using?

**Reply by ahrm** [reply]
Also, how are you opening new files? (e.g. by clicking on them in a gui file browser or command line or some other way?)

**Reply by NailGit** [reply]
My OS is Linux Mint Mate 21 based on Ubuntu 22.04. In the main menu, I created a new item Sioyek, and set the way to open files /home/nail/Progi/Pdf/Sioyek/sioyek/build/sioyek %F . The caja (like Nautilus in Ubuntu) uses this set to open. But mostly I use Double Commander (like Norton Commander) in which I set the path for binary /home/nail/Progi/Pdf/Sioyek/sioyek/build/sioyek and %p to open the file with its whole path. By the way, I tried to copy the binary to /usr/bin and in this case it opened the file but with a completely black window. For this reason, I use binary in the build directory.

**Reply by ahrm** [reply]
What happens if you change the command to open files to this?
```
/path/to/sioyek %F --new-window
```

**Reply by NailGit** [reply]
I have already tried. The same result. 
I'd like to specify the question. Two possibilities. 
1) I opened the first document by using Double Commander (DC). It is open to my preferences. Then I open the second document by using DC. It is opened without my preferences in separate window. 
2) I opened the first document by using Double Commander (DC).  Then I click `o` to open a new document inside an already open window, select another file, and open it. It is opened with my preferences but in the same window, without opening a new window.

**Comment by NailGit** [comment]
> What happens if you change the command to open files to this?
> 
> ```
> /path/to/sioyek %F --new-window
> ```
There is no problem with Appimage binary. I started in the console two times Sioyek-x86_64.AppImage 1.pdf and Sioyek-x86_64.AppImage 1.pdf and both times the file is opened with my preferences. Therefore, the problem connects with building binary by GitHub repository.
